modules/contact/src/constraints/ComputeWeightedGapLMMechanicalContact.C

```diff
@@ -163,11 +169,57 @@ ComputeWeightedGapLMMechanicalContact::computeJacobian(const Moose::MortarType m
   computeResidual(mortar_type);
 }
 
+void
+ComputeWeightedGapLMMechanicalContact::communicateGaps()
+{
+#ifdef MOOSE_SPARSE_AD
```

Comment:
IF this has no sparse AD, will the gaps be correct as well? If not, we might add some messages to inform users

Reply:
We error out in the constructor if the user is using nonsparse

Reply:
Good